external-crates/move/crates/move-compiler/src/typing/ast.rs

```diff
@@ -209,7 +209,17 @@ pub enum UnannotatedExp_ {
         body: Box<Exp>,
     },
     NamedBlock(BlockLabel, Sequence),
-    Block(Sequence),
+    Block(
```

Comment:
@cgswords is suggesting adding a separate node here, something like `MacroInlined, that would wrap a block with the additional metadata and would be generated in the IDE mode. I am cool with that but before making this change I would like to figure out what the best structure would be. Perhaps `MacroInlined(Name, Signature, Sequence)` or `MacroInlined(Name, Signature, Sequence)` ?

Reply:
The best thing to do is `MacroInlined(MacroInfo, Box<Exp>)` where `MacroInfo` is a different `struct` on the side that you can add information to over time. This both helps with extensibility and Rust's stack behavior.

Reply:
I ultimately went for `ExpandedMacro(MacroInfo, Sequence)` as I need direct access to sequence in the analyzer (to extract by-value args that are pre-pended to it (this is needed for the symbols for the receiver in dot-call syntax to be symbolicated correctly). Technically, I could do `ExpandedMacro(MacroInfo, Box<Exp>)` and assert in the IDE that `Box<Exp> is in fact a block, but the overall difference in how the code ultimately looks like in these two variants is not huge and the solution using sequence seems a bit cleaner.